### DISCUSSION TITLE
Custom cloud

### DISCUSSION THREAD

**Discussion by RosData** [NightscoutFoundation/xDrip#3821] [General]
How to set upload to other cloud than Nightscout? I mean, I have custom REST API accepting POST, but when I set the url it does not work.

**Comment by Navid200** [comment]
Please explain what you mean.

xDrip can upload to Nightscout or TidePool or xDrip Sync ...

If you want to upload to Nightscout, these are the instructions: https://navid200.github.io/xDrip/docs/Nightscout-Settings.html

**Reply by RosData** [reply]
I mean I can upload to any API fulfilling the contract for POST /entries . But wait it is working, I set up the url to my own api: http://www.rosdata.cz/api/gluco (accepting also /entries) and I see that json {"device":"xDrip-UiBased","date":1734534365698,"dateString":"2024-12-18T16:06:05.698+0100","sgv":110,"direction":"Flat"} has come. I have NightScout url specific for my son, but the server running nightscout is running by volunteers and sometimes the server is not several hours running, so I will better pay for commercial cloud.

**Reply by RosData** [reply]
What I did is wrote app for M5Core with 2 inch display and I call NS api and show the blood suger, so I need some reliable cloud server.

**Reply by Navid200** [reply]
I don't pay anyone and it works with no interruptions at all: 
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html

**Reply by RosData** [reply]
Looks nice, thank you.

**Reply by akitamostodos** [reply]
Hello, I want to upload data to my server and want to make a api to read it, on upload, on api rest for nightscout I insert my web server example http://server/api.php it upload data, but  if I put http://password@server/api.php.dont upload, can someone help me?

Thanks.

**Reply by Navid200** [reply]
@akitamostodos 
xDrip can upload to Nightscout regardless of where it is served.

If you want to create an alternative to Nightscout, can you explain why Nightscout is not good enough?

**Reply by akitamostodos** [reply]
> @akitamostodos xDrip can upload to Nightscout regardless of where it is served.
> 
> If you want to create an alternative to Nightscout, can you explain why Nightscout is not good enough?

Hello, I use nithscout, but the data is not uploaded by xdrip, I need to upload data from xdrip to server, the flow is, xdrip read from nightscout as follower, and the xdrip can upload data, I want to use this option to upload data to server.

**Reply by Navid200** [reply]
There are many people who use Nightscout and upload to it with an app other than xDrip.  There is nothing wrong with that.
But, you are being evasive and I am not going to insist.

I would like to help you.    If I cannot, I hope that someone else can.  But, the way you have posted your question, I cannot follow.
Hopefully, someone else can help you.

**Reply by akitamostodos** [reply]
> There are many people who use Nightscout and upload to it with an app other than xDrip. There is nothing wrong with that. But, you are being evasive and I am not going to insist.
> 
> I would like to help you. If I cannot, I hope that someone else can. But, the way you have posted your question, I cannot follow. Hopefully, someone else can help you.


Is my first post confuse?, I ask for I need. I need use the field xdrip use to upload to nightscout but this field I use to upload to my own api.
If someone know, want to help me, perfect. I need to do this, no upload to nightscout.

**Reply by Navid200** [reply]
If you use a browser and enter your Nightscout's hostname, does the browser show your Nightscout?

**Reply by RosData** [reply]
If you use xDrip to uload data, but instead of Nightscout you use alternative APi. I did that in following way. on xDrip I set up in NightScout field my own url: http://mySecretUniqueUrl/api/gluco On my server I have to accept also POST to http://mySecretUniqueUrl/api/gluco/entries (xDrip adds /entries automatically). Then xDrip Posts data to my server in json: {"device":"xDrip-UiBased","date":1734887442691,"dateString":"2024-12-22T18:10:42.691+0100","sgv":77,"direction":"Flat"}. Then it is up to you to process the json. Did I answer what you needed:

**Reply by akitamostodos** [reply]
> If you use xDrip to uload data, but instead of Nightscout you use alternative APi. I did that in following way. on xDrip I set up in NightScout field my own url: http://mySecretUniqueUrl/api/gluco On my server I have to accept also POST to http://mySecretUniqueUrl/api/gluco/entries (xDrip adds /entries automatically). Then xDrip Posts data to my server in json: {"device":"xDrip-UiBased","date":1734887442691,"dateString":"2024-12-22T18:10:42.691+0100","sgv":77,"direction":"Flat"}. Then it is up to you to process the json. Did I answer what you needed:

Hello, yes, it is I have and want but the problem is when I put http://passqord@myserverurl/index.php then dosnt save any data if I put http://myserverurl/index.php without authentication it works and save the data you say, and I don't know why or how xdrip send the password. If I do it with curl with password on header it works.

Lot of thanks.

**Reply by RosData** [reply]
I found in xDrip source code: if (uri.getPath().endsWith("/v1/")) apiVersion = 1;  so your api url in xDrip must be  http://passqord@myserverurl/index.php/V1 but xDrip adds /entries so on your server you must have routing for http://passqord@myserverurl/index.php/V1/entries.  I use that without /V1/ so xDrip uses so called legacy format.
